src/drivers/spi/winbond.c

```diff
@@ -197,6 +202,34 @@ static int winbond_write(const struct spi_flash *flash, u32 offset, size_t len,
 	return ret;
 }
 
+static int winbond_sec_read(const struct spi_flash *flash, u32 offset, size_t len, void *buf)
+{
+	int ret = 1;
+	u8 cmd[5];
+	u8 reg = (offset >> 8) & 0xFF;
+	u8 addr = offset & 0xFF;
+
+	if (reg != ADDR_W25_SEC1 && reg != ADDR_W25_SEC2 && reg != ADDR_W25_SEC3) {
+		printk(BIOS_WARNING, "SF: Wrong security register\n");
+		return 1;
+	}
+
+	cmd[0] = CMD_W25_RD_SEC;
+	cmd[1] = 0x0;
+	cmd[2] = reg;
+	cmd[3] = addr;
+	cmd[4] = 0x0; // dummy
+
+	ret = spi_flash_cmd_read(&flash->spi, cmd, sizeof(cmd), buf, len);
+	if (ret) {
+		printk(BIOS_WARNING, "SF: Can't read sec register %d\n", reg >> 4);
```

Comment:
@mek-x isn't `%x` better here ? Do we care about ret value - maybe it can provide failure reason ?

Reply:
@pietrushnic this is just a register number: 1, 2 or 3